Defeat deprecated warnings in php7

Patch by Gordon Bos

--- a/admin/controllers/downloads.php
+++ b/admin/controllers/downloads.php
@@ -4,8 +4,8 @@
 
 	var $sorted=false;
 	var $dl=false;
-	
-	function Downloads(){
+
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -16,8 +16,11 @@
 
 		$this->Auth_model->EnforceAuth('web_admin');
 		$this->Auth_model->DenyUser('admin');
+	}
 
-	}		
+	function Downloads(){
+		self::__construct();
+	}
 
 	function _renderfull($content){
 		$navdata["menu"] = $this->menu->retrieve($this->session->userdata('user'),$this->uri->uri_string());
--- a/admin/controllers/filemanager.php
+++ b/admin/controllers/filemanager.php
@@ -15,9 +15,7 @@
 		return ($a < $b) ? -1 : 1; 
 	}
 
-
-
-	function Filemanager(){
+	function __construct(){
 
 		parent::__construct();
 
@@ -28,6 +26,10 @@
 
 	}
 
+	function Filemanager(){
+		self::__construct();
+	}
+
 	function _renderfull($content, $head=true){
 		if(!is_null($head)) {
             $mdata['head'] = $head;
--- a/admin/controllers/logout.php
+++ b/admin/controllers/logout.php
@@ -2,10 +2,14 @@
 
 class Logout extends CI_Controller{
 
-	function Logout(){
+	function __construct(){
 		parent::__construct();
 	}
 	
+	function Logout(){
+		self::__construct();
+	}
+
 	function index(){
 		$this->Auth_model->Logout();
 		$this->session->unset_userdata("caller");
--- a/admin/controllers/mail.php
+++ b/admin/controllers/mail.php
@@ -2,7 +2,7 @@
 
 class Mail extends CI_Controller{
 
-	function Mail(){
+	function __construct(){
 		parent::__construct();
 		
 		require_once(APPPATH."/legacy/defines.php");
@@ -12,6 +12,10 @@
 
 	}
 
+	function Mail(){
+		self::__construct();
+	}
+
 	function _renderfull($content,$head = "", $mdata){
 		if( ! $head ) {
 			$mdata["head"] = $this->load->view(THEME.'/mail/mail_head_view','',true);
--- a/admin/controllers/music.php
+++ b/admin/controllers/music.php
@@ -2,7 +2,7 @@
 
 class Music extends CI_Controller{
 
-	function Music(){
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -12,6 +12,10 @@
 
 	}
 		
+	function Music(){
+		self::__construct();
+	}
+
 	function index($strip=""){
 		if($strip == "") {
 			$host = explode(":",$_SERVER["HTTP_HOST"]);
--- a/admin/controllers/network.php
+++ b/admin/controllers/network.php
@@ -2,7 +2,7 @@
 
 class Network extends CI_Controller{
 
-	function Network(){
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -14,6 +14,9 @@
 		$this->load->model('system');
 	}
 
+	function Network(){
+		self::__construct();
+	}
 
 	function _renderfull($content, $head = '/network/network_head_view', $data = ''){
 
--- a/admin/controllers/services.php
+++ b/admin/controllers/services.php
@@ -1,7 +1,7 @@
 <?php
 class Services extends CI_Controller{
 	
-	function Services(){
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -11,6 +11,10 @@
 
 	}
 
+	function Services(){
+		self::__construct();
+	}
+
 	function _renderfull($content){
 		$mdata["head"] = $this->load->view(THEME.'/services/services_head_view','',true);
 		$navdata["menu"] = $this->menu->retrieve($this->session->userdata('user'),$this->uri->uri_string());
--- a/admin/controllers/settings.php
+++ b/admin/controllers/settings.php
@@ -2,7 +2,7 @@
 
 class Settings extends CI_Controller{
 
-  function Settings(){
+  function __construct(){
     parent::__construct();
 
     require_once(APPPATH."/legacy/defines.php");
@@ -14,6 +14,10 @@
 
   }
 
+  function Settings(){
+    self::__construct();
+  }
+
   function _renderfull($content,$head=null, $mdata=array()){
 
     $navdata["menu"] = $this->menu->retrieve($this->session->userdata('user'),$this->uri->uri_string());
--- a/admin/controllers/shutdown.php
+++ b/admin/controllers/shutdown.php
@@ -1,7 +1,7 @@
 <?php
 class Shutdown extends CI_Controller{
 
-	function Shutdown(){
+	function __construct(){
 		parent::__construct();
 		
 		require_once(APPPATH."/legacy/defines.php");
@@ -11,6 +11,10 @@
 
 	}
 	
+	function Shutdown(){
+		self::__construct();
+	}
+
 	function index($strip=""){
 		confirm();	
 	}
--- a/admin/controllers/upload.php
+++ b/admin/controllers/upload.php
@@ -2,7 +2,7 @@
 
 class Upload extends CI_Controller{
 	
-	function Upload(){
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -12,6 +12,10 @@
 
 	}		
 
+	function Upload(){
+		self::__construct();
+	}
+
 	function index(){
 		$data["path"]="/".join("/",array_slice($this->uri->segment_array(),2));
 		$this->load->view(THEME.'/upload/upload_index_view',$data);
--- a/admin/controllers/users.php
+++ b/admin/controllers/users.php
@@ -2,7 +2,7 @@
 
 class Users extends CI_Controller{
 
-	function Users(){
+	function __construct(){
 		parent::__construct();
 
 		require_once(APPPATH."/legacy/defines.php");
@@ -11,6 +11,10 @@
 
 	}
 
+	function Users(){
+		self::__construct();
+	}
+
 	private function _renderfull($content, $head = ''){
 		if( ! $head ) {
 			$mdata["head"] = $this->load->view(THEME.'/users/user_head_view','',true);
--- a/admin/models/auth_model.php
+++ b/admin/models/auth_model.php
@@ -83,7 +83,8 @@
 	private $_xml_file = "/etc/bubba/auth.xml";
 	private $_lock_file = "/var/lock/bubba_auth.lock";
 
-	function Auth_model(){
+	public function __construct()
+	{
 		parent::__construct();
 		$this->_dom = new DOMDocument();
 		$this->_dom->formatOutput = true;
@@ -91,6 +92,10 @@
 		$this->_dom->encoding ='utf-8';		
 	}
 
+	function Auth_model(){
+		self::__construct();
+	}
+
 	function Auth($username, $password){
 		$auth = false;
 		$this->groups = array();
--- a/admin/models/menu.php
+++ b/admin/models/menu.php
@@ -430,7 +430,7 @@
             && isset($selected_children)
             && is_array($selected_children)
         ) {
-            $selected['children'] = &$this->_retrieve( $selected_children, $user, $current_level );
+            $selected['children'] = $this->_retrieve( $selected_children, $user, $current_level );
         }
         unset( $selected );
         unset( $selected_children );
--- a/admin/libraries/Session.php
+++ b/admin/libraries/Session.php
@@ -28,14 +28,17 @@
 
 	var $flash_key = 'flash'; // prefix for "flash" variables (eg. flash:new:message)
 
-	function CI_Session()
-	{
+	function __construct(){
 		$this->object =& get_instance();
 		log_message('debug', "Native_session Class Initialized");
 		$this->_sess_run();
 	}
 
-	/**
+	function CI_Session(){
+		self::__construct();
+	}
+
+   /**
     * Regenerates session id
     */
 	function regenerate_id()
--- a/admin/models/disk_model.php
+++ b/admin/models/disk_model.php
@@ -31,7 +31,7 @@
 
 	// snagged from http://se.php.net/manual/en/function.array-multisort.php#89918
 	// _array_sort(string $field, [options, ], string $field2, [options, ], .... , $array)
-	function array_sort(&$array /*, ...*/) {
+	function array_sort($array /*, ...*/) {
 		$args  = func_get_args();
 		$array = array_shift($args);
 		if (! is_array($array)) return false;

