
diff -u a/system/core/Output.php b/system/core/Output.php
--- a/system/core/Output.php
+++ b/system/core/Output.php
@@ -456,7 +456,7 @@
 
 		$elapsed = $BM->elapsed_time('total_execution_time_start', 'total_execution_time_end');
 
-		if ($this->parse_exec_vars === TRUE)
+		if ($this->parse_exec_vars === TRUE && $output)
 		{
 			$memory	 = ( ! function_exists('memory_get_usage')) ? '0' : round(memory_get_usage()/1024/1024, 2).'MB';
 
diff --git a/system/core/Loader.php b/system/core/Loader.php
index 648b7cfc7c2..32f3873262d 100644
--- a/system/core/Loader.php
+++ b/system/core/Loader.php
@@ -49,6 +49,7 @@
  * @author		EllisLab Dev Team
  * @link		https://codeigniter.com/userguide3/libraries/loader.html
  */
+#[AllowDynamicProperties]
 class CI_Loader {
 
 	// All these are set automatically. Don't mess with them.
diff --git a/system/core/Router.php b/system/core/Router.php
index e0fb922f1a6..8302dc06006 100644
--- a/system/core/Router.php
+++ b/system/core/Router.php
@@ -58,6 +58,13 @@ class CI_Router {
 	 */
 	public $config;
 
+	/**
+	 * CI_URI class object
+	 *
+	 * @var	object
+	 */
+	public $uri;
+
 	/**
 	 * List of routes
 	 *
diff --git a/system/core/URI.php b/system/core/URI.php
index 1e948588354..95b9c926c96 100644
--- a/system/core/URI.php
+++ b/system/core/URI.php
@@ -51,6 +51,13 @@
  */
 class CI_URI {
 
+	/**
+	 * CI_Config instance
+	 *
+	 * @var	CI_Config
+	 */
+	public $config;
+
 	/**
 	 * List of cached URI segments
 	 *

