
diff -u a/system/core/Output.php b/system/core/Output.php
--- a/system/core/Output.php
+++ b/system/core/Output.php
@@ -456,7 +456,7 @@
 
 		$elapsed = $BM->elapsed_time('total_execution_time_start', 'total_execution_time_end');
 
-		if ($this->parse_exec_vars === TRUE)
+		if ($this->parse_exec_vars === TRUE && $output)
 		{
 			$memory	 = ( ! function_exists('memory_get_usage')) ? '0' : round(memory_get_usage()/1024/1024, 2).'MB';
 

