Fix various issues with networking

Patch by Gordon Bos

--- web-admin/bin/adminfunctions.php
+++ web-admin/bin/adminfunctions.php
@@ -467,6 +467,8 @@
 	return $res;
 }
 
+/*
+	// Gordon: 2015-07-15 - does not appear to be used any more
 function _check_dhcp($iface=""){
 	if($iface=""){
 		$iface=_getlanif();
@@ -543,6 +545,7 @@
 	return $res;	
 	
 }
+*/
 
 function set_static_netcfg($iface, $ip,$nm,$gw){
    
--- web-admin/lib/Bubba.pm
+++ web-admin/lib/Bubba.pm
@@ -532,6 +532,28 @@
 
 }
 
+# Gordon : 2015-06-22 - added function to keep existing hosts entries
+sub update_hostsfile {
+
+	my ($lanip,$name) = @_;	
+	if ($lanip=="127.0.0.1") { return 0;}
+	my $oldname;
+	use File::Slurp;
+	my $hosts = read_file('/etc/hosts');
+	my @lines = split("\n", $hosts);
+	chomp(@lines);
+	foreach (@lines) {
+		if ( /^$lanip(.+)$/ ) {
+			/\s([^\s\.]+)[\s\.]/;
+			$oldname=$1;
+		}
+	}
+	$hosts =~ s/\$oldname/$lanip/g;
+	write_file( '/etc/hosts', $hosts );
+
+}
+
+
 # Change hostname
 #
 # Args   : name - New hostname  
@@ -544,14 +566,15 @@
 	my ($name)=@_;
 
 	system("echo $name > /proc/sys/kernel/hostname");
-	system("echo $name > /etc/hostname");
-	system("echo $name.localdomain > /etc/mailname");
+	system("sed -i \"s/\s*\(hostname=\).*$/\\1\\\"$name\\\"/\"   /etc/conf.d/hostname");
+#	system("echo $name.localdomain > /etc/mailname");
 
 	%ifs = read_interfaces();
 	chomp($lan = _get_lanif);
 	$lanip = $ifs{$lan}{"options"}{"address"};
 	$lanip = '127.0.0.1' unless $lanip;
-	write_hostsfile($lanip,$name);
+#	write_hostsfile($lanip,$name);
+	update_hostsfile($lanip,$name);
 
 	if(!query_service("dnsmasq")){
 		#restart dnsmasq
@@ -564,10 +587,10 @@
 	system("echo send host-name \\\"$name\\\"\\\; >> /etc/dhcp/dhclient.conf.new");
 	system("mv /etc/dhcp/dhclient.conf.new /etc/dhcp/dhclient.conf");
 	$lan = _get_lanif;
-	system("/sbin/ifup --force eth0 $lan");
+	system("rc-config restart `rc-config list default | grep "^\s*net\."`");
 
 	if(change_ftp_servername($name)){
-		system("/etc/init.d/proftpd restart");
+		system("rc-config restart proftpd");
 	}
 
 	restart_avahi();
@@ -639,9 +662,9 @@
    my ($if)=@_;
    my $ret;
 
-   $ret=system("/sbin/ifdown $if");
+   $ret=system("rc-config stop net.$if");
    if ($ret==0) {
-      $ret=system("/sbin/ifup $if");
+      $ret=system("rc-config start net.$if");
    }
    return $ret;
 }
@@ -769,7 +792,25 @@
 sub set_nameserver{
 	my ($ns)=@_;
 	
-	return system("echo -ne 'search\nnameserver $ns\n'>/etc/resolv.conf");
+#	return system("echo -ne 'search\nnameserver $ns\n'>/etc/resolv.conf");
+# Gordon : 2015-06-22 Don't delete domain information in this file
+	my $findstring="nameserver";
+	my $ret;
+        use File::Slurp;
+        my $resolvconf = read_file('/etc/resolv.conf');
+        my @lines = split("\n", resolvconf);
+        chomp(@lines);
+        foreach (@lines) {
+                if ( /^$findstring/ ) {
+                        $ret .= "nameserver $ns\n";
+			$findstring="-";
+                }else {
+			$ret .= $_"\n";
+		}
+        }
+        
+        write_file( '/etc/resolv.conf', $ret );
+
 }
 
 # Is service running?

