diff --git a/Slim/Utils/Misc.pm b/Slim/Utils/Misc.pm
index fca9db1..3d9f7f4 100644
--- a/Slim/Utils/Misc.pm
+++ b/Slim/Utils/Misc.pm
@@ -852,15 +852,7 @@ sub fileFilter {
 	elsif ($validRE && -f _) {
 		return 0 if $item !~ $validRE;
 	}
-	elsif ($validRE && -l _ && defined ($target = readlink($fullpath))) {
-		# fix relative/absolute path
-		$target = ($target =~ /^\// ? $target : catdir($dirname, $target));
 
-		if (-f $target) {
-			return 0 if $target !~ $validRE;
-		}
-	}
-	
 	return 1;
 }
 
