diff -ru a/sakaki-tools/app-portage/showem/Manifest b/sakaki-tools/app-portage/showem/Manifest
--- a/sakaki-tools/app-portage/showem/Manifest
+++ b/sakaki-tools/app-portage/showem/Manifest
@@ -1 +1 @@
-DIST showem-1.0.3.tar.gz 16581 SHA256 3b980b7ab41d531e062a338cb2ac3753ba9f31f0b5df5bf2a439ead2786934a7 SHA512 333dd83e98c028618996ba6b392d419bf6b9399200e2d22952c36da6ae57793d521edef0c16dfe01b0adb5c106e969b5288e89514c59f0a36cfe1e4e654aab4d WHIRLPOOL 759a197373e18c9c9f73f442edcbb58ed3564083581f4ddc4240ff208e654bd65be9e5e87008d40a4759a91c6a77fc012ab6717f949ae9c841b5572a08cbbcf4
+DIST showem-1.0.3.tar.gz 16581 BLAKE2B acbb043c05dc1b1bd6f28036e813cd9069d5a7f88da5b90707f58170100761397e56b83fb292e31e5487469b8ed202f096deea79e42c736a23c803b0e36a71d3 SHA512 333dd83e98c028618996ba6b392d419bf6b9399200e2d22952c36da6ae57793d521edef0c16dfe01b0adb5c106e969b5288e89514c59f0a36cfe1e4e654aab4d
diff -ru a/sakaki-tools/app-portage/showem/showem-1.0.3.ebuild b/sakaki-tools/app-portage/showem/showem-1.0.3.ebuild
--- a/sakaki-tools/app-portage/showem/showem-1.0.3.ebuild
+++ b/sakaki-tools/app-portage/showem/showem-1.0.3.ebuild
@@ -5,8 +5,6 @@
 
 EAPI="8"
 
-inherit eutils
-
 DESCRIPTION="View output of a parallel emerge from a separate terminal"
 BASE_SERVER_URI="https://github.com/sakaki-"
 HOMEPAGE="${BASE_SERVER_URI}/${PN}"
@@ -24,9 +22,6 @@
 	>=app-shells/bash-4.2"
 
 # ebuild function overrides
-src_prepare() {
-	epatch_user
-}
 src_install() {
 	dobin "${PN}"
 	doman "${PN}.1"
