Netifrc/Brctl migration to sysfs

Patch by Gordon Bos

--- a/utils/InterfacesCfg.cpp
+++ b/utils/InterfacesCfg.cpp
@@ -73,6 +73,8 @@
 		if(line.substr(0,7)=="bridge_"){
 			size_t argsep=line.find_first_of("=");
 			curif=line.substr(7,(argsep-7));
+			if(curif.size()>7)	// sysfs entry
+				continue;
 			arguments=line.substr(argsep+1);
 			isbridge=true;
 		}
@@ -184,6 +186,8 @@
 			size_t argsep=line.find_first_of("=");
 			arguments=line.substr(argsep+1);
 			curif=line.substr(7,(argsep-7));
+			if(curif.size()>7)	// sysfs entry
+				continue;
 			ignoreline=true;
 		}
 		if(line.substr(0,6)=="brctl_"){
@@ -256,9 +260,9 @@
 					ss<<endl;
 					res.push_back("bridge_"+curif+"=\"");
 					res.push_back(ss.str());
-					res.push_back("brctl_"+curif+"=\"setfd 0\n");
-					res.push_back("sethello 10\n");
-					res.push_back("stp on\"\n");
+					res.push_back("bridge_forward_delay_"+curif+"=0\n");
+					res.push_back("bridge_hello_time_"+curif+"=1000\n");
+					res.push_back("bridge_stp_state_"+curif+"=1\n");
 				}
 				if(devs[curif]["addressing"]=="dhcp"){
 					res.push_back("config_"+curif+"=\"dhcp\"\n");
